nornir_nautobot/plugins/tasks/dispatcher/default.py

```diff
@@ -241,9 +242,7 @@ def get_config(task: Task, logger, obj, backup_file: str, remove_lines: list, su
                 { "config: <running configuration> }
         """
         logger.log_debug(f"Executing get_config for {task.host.name} on {task.host.platform}")
-        command = NautobotNornirDriver.RUN_COMMAND_MAPPING.get(
-            task.host.platform, NautobotNornirDriver.RUN_COMMAND_MAPPING["default"]
-        )
+        command = NetmikoNautobotNornirDriver.config_command
```

Comment:
command = self.config_command

Reply:
@itdependsnetworks same as PR #76 , these are class attrs so we need to reference them as cls.(). Unless all methods are turned into class methods. 
self.() references an instance attr, but drivers implement actions as static methods, so that wouldn't work.

Example::

```
Type "help", "copyright", "credits" or "license" for more information.
>>> class parentClass:
...     parentAttr = "parentAttr"
... 
>>> class childClass(parentClass):
...     childAttr = "childAttr"
...     @staticmethod
...     def printAttr():
...             print(self.childAttr)
... 
>>> childClass.printAttr()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 5, in printAttr
NameError: name 'self' is not defined
```

Reply:
Dog’!! 

Ok, let me think about it

Reply:
```
>>> class parentClass:
...     parentAttr = "parentAttr"
...
>>> class childClass(parentClass):
...     childAttr = "childAttr"
...     @staticmethod
...     def printAttr():
...             print(__class__.childAttr)
...
>>> childClass.printAttr()
childAttr
>>>
>>>
>>>
>>> class childClass(parentClass):
...     childAttr = "childAttr"
...     @staticmethod
...     def printAttr():
...             print(__class__.parentAttr)
...
>>> childClass.printAttr()
parentAttr
>>>
```

Reply:
correct, from a child class we can access both attrs with __class__ dunder. In our case we override the parent attr (config_command) in the scope/namespace of the child class:

```
>>> class childClass(parentClass):
...     childAttr = "this is a child class"
...     @staticmethod
...     def printAttr():
...             print(childClass.childAttr)

>>> dir(childClass)
['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', 'childAttr', 'parentAttr', 'printAttr']


>>> class anotherchildClass(parentClass):
...     parentAttr = "this is not a  child"
...     @staticmethod
...     def printAttr():
...             print(childClass.parentAttr)

>>> dir(anotherchildClass)
['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', 'parentAttr', 'printAttr']

```

in both cases, we refer to the same python object:

```
>>> class anotherchildClass(parentClass):
...     parentAttr = "this is not a  child"
...     @staticmethod
...     def printAttr():
...             print(id(__class__.parentAttr))
... 

>>> id(anotherchildClass.parentAttr)
139800943187872
>>> anotherchildClass.printAttr()
139800943187872

```

do you want to switch to __class__ instead?

Reply:
Yes, sorry, I should have been more explicit, that wasn’t just an example but a way I thought made sense to handle this situation

Reply:
So we need to find a way to make it work then for classes that inherit methods from parents, like Cisco ones, as this approach will have some issues in these cases:

```>>> class ParentClass:
...     config_command = "some_command"
...     @staticmethod
...     def do_something():
...             print(__class__.config_command)
... 
>>> class ChildClass(ParentClass):
...     config_command = "some_other_command"
... 
>>> ParentClass.do_something()
some_command
>>> ChildClass.do_something()
some_command
```

`__class__` will look into the namespace of the class where the method is invoked (Parent) and therefore use that local class attr. Cisco ones will be fine, but we'll break junos for instance. Need to think how to make that work.

Reply:
perhaps I misunderstand the issue, in my mind this is all of the cases we have to consider. 

```
>>> class parentClass:
...     firstAttr = "firstAttr"
...     secondAttr = "secondAttr"
...
>>>
>>> class childClass(parentClass):
...     firstAttr = "firstAttr-override"
...     thirdAttr = "thirdAttr"
...     @staticmethod
...     def printAttr1():
...             print(__class__.firstAttr)
...     @staticmethod
...     def printAttr2():
...             print(__class__.secondAttr)
...     @staticmethod
...     def printAttr3():
...             print(__class__.thirdAttr)
...
>>>
>>> childClass.printAttr1()
firstAttr-override
>>> childClass.printAttr2()
secondAttr
>>> childClass.printAttr3()
thirdAttr
>>>
```